packages/tx/src/capabilities/legacy.ts

```diff
@@ -45,14 +45,16 @@ export function hash(tx: LegacyTxInterface): Uint8Array {
     throw new Error(msg)
   }
 
+  const keccackFunction = tx.common.customCrypto.keccak256 ?? keccak256
+
   if (Object.isFrozen(tx)) {
     if (!tx.cache.hash) {
-      tx.cache.hash = keccak256(tx.serialize())
+      tx.cache.hash = keccackFunction(tx.serialize())
```

Comment:
Really happy here that our capability consolidation in tx is already paying off. 🤩

//cc @gabrocheleau